src/CoordinateSubject.java

```diff
@@ -10,6 +10,8 @@
 	//
 	public void notifyObserver() throws Exception {
 
+		attach(observer);
```

Comment:
Notice that this project utilizes the `this` keyword in contexts where it's not always needed. It's best to be consistent throughout the entire codebase.